Fix `aggregate` and `distinct` Methods Bugs

ThisaruGuruge opened the pull request:
## Purpose

- Fixed the issue where the `aggregate` method cannot handle union types
- Fixed the issue where the `distinct` method fails due to value conversion error for complex types
- Added texts for additional edge cases
- Added new keywords for the package
- Renamed the `Package.md` to `README.md` and removed the `Module.md`

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/8163
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/8164

## Examples

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~Updated the spec~
- [x] Checked native-image compatibility

niveathika commented on native/src/main/java/io/ballerina/lib/mongodb/IteratorUtils.java:
```diff
@@ -49,16 +49,20 @@ public static Object nextResult(BObject iterator) {
                 BTypedesc nextValueTypeDesc = ValueCreator.createTypedescValue(nextValueType);
                 return FromJsonStringWithType.fromJsonStringWithType(fromString(result), nextValueTypeDesc);
+            } catch (BError e) {
+                return ErrorCreator.createError(fromString("Failed to convert value to the expected type"), e);
```
Shall we print the value an expected type?

TharmiganK commented:
The `5.1.0` version is released with Ballerina SwanLake Update 11(2201.11.0) - https://github.com/ballerina-platform/module-ballerinax-mongodb/blob/f8f0eb2aecb5dcecdd3098b724d44586e426572d/gradle.properties#L4
So if we are publishing this with Update 12, we need to bump to the next minor version right?

Also the lang version in the master is `2201.12.0`, so if we want this fixes for Update 11, then we need to release a patch on update 11 right?

ThisaruGuruge commented:
> The `5.1.0` version is released with Ballerina SwanLake Update 11(2201.11.0) -
> 
> https://github.com/ballerina-platform/module-ballerinax-mongodb/blob/f8f0eb2aecb5dcecdd3098b724d44586e426572d/gradle.properties#L4

Updated to Update 12 now.

> So if we are publishing this with Update 12, we need to bump to the next minor version right?
> Also the lang version in the master is `2201.12.0`, so if we want these fixes for Update 11, then we need to release a patch on update 11 right?

They want this on Update 12. Therefore bumped to the next minor version and updated the PR.

niveathika reviewed (approved):
LGTM
